server/startup/restApiEndpoints.js

```diff
@@ -0,0 +1,188 @@
+import collections from "../../lib/collections";
+import Reaction from "/server/api/core";
+
+
+/**
+ *
+ * @param {object} user
+ * @param {string} role
+ * @returns {boolean}
+ */
+const hasPermission = (user, role) => {
+  return user.roles[Reaction.getShopId()].includes(role);
+};
+
+const Api = new Restivus({
+  apiPath: "api",
+  useDefaultAuth: true,
+  prettyJson: true,
+  version: "v1",
+  defaultHeaders: {
+    "Content-Type": "application/json"
+  }
+});
+/**
+ *
+ *
+ * @param {String} collectionName -Name of collection
+ * @param {Object} collection - collection Schema
+ * @param {Object} restivus object
+ * @returns none
+ */
+function creatRestApiFor(collectionName, collection, restApi = Api) {
+  restApi.addCollection(collection, {
+    routeOptins: { authRequired: true },
+    endpoints: {
+      getAll: {
+        authRequired: true,
+        action() {
+          if (hasPermission(this.user, "admin")
+            || hasPermission(this.user, "guest")
+            || hasPermission(this.user, "owner")) {
+            const all = collection.find().fetch();
+            if (!all) {
+              return {
+                status: "failure",
+                statusCode: 404,
+                message: `No ${collectionName} found!`
+              };
+            }
+            return {
+              status: "sucess",
+              statusCode: 200,
+              data: {
+                collectionName: all
+              }
+            };
+          }
+          return {
+            statusCode: 403, status: "fail",
```

Comment:
Be consistent with your response status when the request fails. on line 45 your status is 'failure', line 59 'fail' and line 79 'failed'. choose one and be consistent